File: src/main/java/hacker/solution/WordSplit.java
package hacker.solution;

import java.util.Scanner;

public class WordSplit {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        String s = scan.nextLine();
        // Write your code here.
        String regex = "\\P{Alpha}+";
        String[] tokens = s.split(regex);
        int count = 0;
        for (int i = 0; i < tokens.length; i++) {
            String token = tokens[i];
            token = token.trim();
            if (token.length() == 0) {
                count++;
            }
            if (token.contains("'")) count--;
        }
        System.out.println(tokens.length - count);
        for (String element : tokens) {
            if (element.trim().length() > 0) {
                if (element.contains("'")) {
                    String[] elems = element.split("'");
                    for (String elem : elems) {
                        System.out.println(elem);
                    }
                } else System.out.println(element);
            }
        }
        scan.close();
    }

}

File: src/main/kotlin/hacker/solutionk/WordSplit.kt
package hacker.solutionk

import java.util.*

    fun main(args: Array<String>) {
        val scan = Scanner(System.`in`)
        val s = scan.nextLine()
        // Write your code here.
        val regex = "\\P{Alpha}+"
        val tokens = s.split(regex.toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        var count = 0
        for (i in tokens.indices) {
            var token = tokens[i]
            token = token.trim { it <= ' ' }
            if (token.length == 0) {
                count++
            }
            if (token.contains("'")) count--
        }
        println(tokens.size - count)
        for (element in tokens) {
            if (element.trim { it <= ' ' }.length > 0) {
                if (element.contains("'")) {
                    val elems = element.split("'".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                    for (elem in elems) {
                        println(elem)
                    }
                } else
                    println(element)
            }
        }
        scan.close()
    }

